src/components/Calculator.js

```diff
@@ -1,17 +1,30 @@
 import React, { Component } from 'react';
 import Button from './Button';
+import calculate from '../logic/calculate';
 
 class Calculator extends Component {
   constructor() {
     super();
     this.buttons = ['AC', '+/-', '%', '÷', '7', '8', '9', 'x', '4', '5', '6', '-', '1', '2', '3', '+', '0', '.', '='];
+    this.state = {
+      total: '0',
+      next: null,
+      operation: null,
+    };
+  }
+
+  onClickButton = (obj, text) => {
+    const newobj = calculate(obj, text);
+    if (newobj.total === null) newobj.total = '0';
+    this.setState(newobj);
   }
 
   render() {
+    const { total } = this.state;
     return (
       <section className="mainframe">
-        <p className="result">0</p>
-        {this.buttons.map((text) => <Button content={text} key={`btn-${text}`} />)}
+        <p className="result">{ total }</p>
+        {this.buttons.map((text) => <Button content={text} key={`btn-${text}`} click={() => this.onClickButton(this.state, text)} />)}
       </section>
```

Comment:
- [ ] No button(click) is working for me on the live version of the project. Please make sure the logic works fine for this project.

Reply:
As I asked in the original post, could you explain the expected behaviour? Could you explain what is it that doesn't work?

Reply:
Hi there , 
It should work just like a calculator but for your project I cannot use button. It's mean handle click has a problem.